### ISSUE TITLE
host_vars for new hosts via add_host are not loaded anymore

### ISSUE THREAD

**Issue by illenseer** [ansible/ansible#10500]
Currently we are using `add_host`, when we automatically build templates and it uses the variables provided in a `host_vars` file. This works up to 1.7.2, but in 1.8.x the variables from the `host_vars` file are gone.

This is what 1.7.2 delivers:

```
$ ansible-playbook --version; ansible-playbook -i inventory test.yml
ansible-playbook 1.7.2

PLAY [localhost] **************************************************************

TASK: [add_host name=test groups=test] ****************************************
ok: [localhost]

TASK: [debug var=hostvars['test']] ********************************************
ok: [localhost] => {
    "hostvars['test']": {
        "group_names": [
            "test"
        ],
        "inventory_hostname": "test",
        "inventory_hostname_short": "test",
        "test_group_vars": "group test value",
        "test_host_vars": "host test value"
    }
}

PLAY RECAP ********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0
```

In 1.8.x the variables defined in `host_vars` are gone:

```
$ ansible-playbook --version; ansible-playbook -i inventory test.yml
ansible-playbook 1.8.4
  configured module search path = None

PLAY [localhost] **************************************************************

TASK: [add_host name=test groups=test] ****************************************
ok: [localhost]

TASK: [debug var=hostvars['test']] ********************************************
ok: [localhost] => {
    "hostvars['test']": {
        "group_names": [
            "test"
        ],
        "inventory_hostname": "test",
        "inventory_hostname_short": "test",
        "test_group_vars": "group test value"
    }
}

PLAY RECAP ********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0
```

This is the example used to show the problem:

```
$ tree
.
├── inventory
│   ├── group_vars
│   │   └── test.yml
│   ├── host_vars
│   │   └── test.yml
│   └── hosts
└── test.yml
```

``` yaml
# inventory/group_vars/test.yml:

---
test_group_vars: group test value
```

``` yaml
# inventory/host_vars/test.yml:

---
test_host_vars: host test value
```

``` ini
# inventory/hosts:
localhost
```

``` yaml
# test.yml:

---
- hosts: localhost
  gather_facts: False
  tasks:
    - add_host: name=test
                groups=test
    - debug: var=hostvars['test']
```

**Comment by bcoca** [comment]
have you tested against current development?

**Comment by bcoca** [comment]
bisect found the issue was introduced at a2cfe8770b9a93ed2a1372396fa7a1e648bd658e

**Comment by bcoca** [comment]
@srvg could you take a look? you probably know this code better than i do

**Comment by bcoca** [comment]
please check patch above to see if it fixes the issue for you

**Comment by illenseer** [comment]
Okay, I checked development with and without patch and 1.8.4 with patch, but `host_vars` are still not loaded.

**Comment by bcoca** [comment]
ok, it seems fact caching was giving me a red herring, try now

**Comment by illenseer** [comment]
This patch fixes the issue and for devel and 1.8.4 `host_vars` are correctly loaded.
Funny thing, I just made a nearly identical patch :-)
